src/TriggerActivityMakerTrackFinding.cpp

```diff
@@ -0,0 +1,303 @@
+
+  unsigned int channel = 0;      // Current channel ID
+  unsigned int next_channel = 0; // Next channel ID
+  unsigned int next = 0;         // The next position in the hit channels vector
+  unsigned int tol_count = 0;    // Tolerance count, should not pass adj_tolerance
+
+  // Generate a channelID ordered list of hit channels for this window; second element of pair is tps
+  std::vector< std::pair <int, TriggerPrimitive> > chanTPList;
+  for (auto tp : m_current_window.inputs) {
+    chanTPList.push_back( std::make_pair(tp.channel, tp) );
+  }
+  std::sort(chanTPList.begin(), chanTPList.end(), [](const std::pair<int,TriggerPrimitive> &a, const std::pair<int,TriggerPrimitive> &b){return (a.first < b.first);});
+  
+  // ADAJACENCY LOGIC ====================================================================
+  // =====================================================================================
+  // Adjcancency Tolerance = Number of times prepared to skip missed hits before resetting 
+  // the adjacency count (win_adj). This accounts for things like dead channels / missed TPs. The 
+  // maximum gap is 4 which comes from tuning on December 2021 coldbox data, and June 2022 
+  // coldbox runs.
+
+  TPWindow win_adj;              // add first tp, and then if tps are on next channels (check code below to understand the definition)
+  TPWindow win_adj_max;          // if track length > m_adjacency_threshold, set win_adj_max = win_adj; return win_adj_max;
+  
+  for (int i = 0; i < chanTPList.size(); ++i) {
+
+    win_adj_max.clear();
+
+    next = (i + 1) % chanTPList.size();       // Loops back when outside of channel list range
+    channel = chanTPList.at(i).first;
+    next_channel = chanTPList.at(next).first; // Next channel with a hit
+
+    // End of vector condition.
+    if (next == 0) { next_channel = channel - 1; }
+    
+    // Skip same channel hits.
+    if (next_channel == channel) continue;
+
+    // If win_adj size == zero, add current tp
+    if (win_adj.inputs.size() == 0) win_adj.add(chanTPList[i].second);
+
+    // If next hit is on next channel, increment the adjacency count
+    if (next_channel - channel == 1) {
+      win_adj.add(chanTPList[next].second);
+    }
+    
+    // Allow a max gap of 4 channels (e.g., 45 and 50; 46, 47, 48, 49 are missing); increment the adjacency count
+    // Sum of gaps should be < adj_tolerance (e.g., if toleance is 30, the max total gap can vary from 0 to 29+4 = 33)
+    else if (next_channel - channel > 0 && next_channel - channel <= 5 &&
+	     tol_count < m_adj_tolerance) {
+      win_adj.add(chanTPList[next].second);
+      for (int i = 0; i < next_channel-channel-1; ++i) ++tol_count;
+    }
+
+    // if track length > m_adjacency_threshold, set win_adj_max = win_adj;
+    else if (win_adj.inputs.size() > m_adjacency_threshold) {
+      win_adj_max = win_adj;
+      break;
+    }
```

Comment:
in case `win_adj.inputs.size() > m_adjacency_threshold` is not satisfied, `return win_adj;` will not execute. In case this condition is not satisfied, the `win_adj` is cleared and search for another activity continues.